test/reductions/test_grid_search_classification.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from fairlearn.metrics.demographic_parity import DemographicParity
+from fairlearn.reductions.grid_search_classification import GridSearchClassification
+
+import numpy as np
+import pandas as pd
+from sklearn.linear_model import LogisticRegression
+
+
+class TestGridSearchClassification:
+    def _simple_threshold_data(self,
+                               number_a0, number_a1,
+                               a0_threshold, a1_threshold,
+                               a0_label, a1_label):
+
+        a0s = np.full(number_a0, a0_label)
+        a1s = np.full(number_a1, a1_label)
+
+        a0_scores = np.linspace(0, 1, number_a0)
+        a1_scores = np.linspace(0, 1, number_a1)
+        score_feature = np.concatenate((a0_scores, a1_scores), axis=None)
+
+        A = np.concatenate((a0s, a1s), axis=None)
+
+        Y_a0 = [x > a0_threshold for x in a0_scores]
+        Y_a1 = [x > a1_threshold for x in a1_scores]
+
+        Y = np.concatenate((Y_a0, Y_a1), axis=None)
+
+        X = pd.DataFrame({"actual_feature": score_feature,
+                          "protected_attribute_feature": A,
+                          "constant_ones_feature": np.ones(len(Y))})
+        return X, Y, A
+
+    def test_demographicparity_fair_uneven_populations(self):
+        # Variant of test_demographicparity_already_fair, which has unequal
+        # populations in the two classes
+        # Also allow the threshold to be adjustable
+
+        score_threshold = 0.625
+
+        number_a0 = 4
+        number_a1 = 4
+
+        a0_label = 17
+        a1_label = 37
+
+        X, Y, A = self._simple_threshold_data(number_a0, number_a1,
+                                              score_threshold, score_threshold,
+                                              a0_label, a1_label)
+
+        target = GridSearchClassification(LogisticRegression(),
```

Comment:
Have started using sklearn since the existing `simple_learner` wasn't quite giving its predictions as I expected.